create.ps1

```diff
@@ -44,16 +45,33 @@ try{
             $PortalBaseUrl = $PortalBaseUrl + "/"
         }
         $uri = ($PortalBaseUrl +"api/v1/users/")
-
+    #Append desired username to portal uri
+    $correlationUri = ($uri + "$($account.username)")
+    #Look for account by username
+    $correlationResponse = Invoke-RestMethod -Method GET -Uri $correlationUri -Headers $headers
+    #Correlate User
+    $aRef = $correlationResponse.userGUID
+    $success = $True;
+    $auditMessage = " $($p.DisplayName) found and correlated successfully"
+    }
+}catch{
+    #User not found (expected case)
+    if($error[0].Exception -like "*404*"){
```

Comment:
What happens when the user is found? Seems to me that we need to return the user GUID from the correlationResponse in case the user is found. This so we don't get any failed entitlements and memberships. Do you agree with me?

Reply:
Line 53, the userGUID is set as the account reference.

Reply:
Line 53, the userGUID is set as the account reference.

Reply:
My mistake I missed that. It's good.